plotly/plotly/plotly.py

```diff
@@ -1308,7 +1308,19 @@ def response_handler(cls, response):
                     .format(url=get_config()['plotly_api_domain'])
                 )
             else:
-                raise requests_exception
+                try:
+                    parsed_response = response.json()
+                except:
+                    parsed_response = response.content
+
+                try:
+                    # try get a friendly error message
+                    errors = parsed_response.get('errors')
+                    message = errors[-1]['message']
+                except:
+                    raise requests_exception
+
+                raise exceptions.PlotlyError(message)
```

Comment:
aw jeez. i didn't realize it was still in a try block. that's fine then. But, ya that's what I was talking about. The lines I was linking to were:

```
    try:
        parsed_response = r.json()
    except:
        parsed_response = r.content

    # raise error message
    if not r.ok:
        message = ''
        if isinstance(parsed_response, dict):
            errors = parsed_response.get('errors')
            if errors and errors[-1].get('message'):
                message = errors[-1]['message']
        if message:
            raise exceptions.PlotlyError(message)
        else:
            # shucks, we're stuck with a generic error...
            r.raise_for_status()
```